app/services/foreman_discovery/host_converter.rb

```diff
@@ -19,6 +22,41 @@ def self.to_managed(original_host, set_managed = true, set_build = true, added_a
     host
   end
 
+  def self.eventually_make_bond(host)
+    primary = host.primary_interface
+    return if primary.nil?
+    return if primary.type == 'Nic::Bond'
+
+    neighbors = ::ForemanDiscovery::LldpNeighbors
+      .from_facts(host.facts_hash)
+      .get_neighbors_by_interface(primary.identifier)
+
+    return if neighbors.nil?
+
+    ip = primary.ip
+    name = primary.name
+    primary.update(
+       :primary   => false,
+       :provision => false,
+       :managed   => false,
+       :name      => nil,
+       :ip        => nil
+    )
+
+    bond = Nic::Bond.create(
+        :identifier       => "bond0",
+        :attached_devices => neighbors,
+        :primary          => true,
+        :provision        => true,
+        :name             => name,
+        :ip               => ip,
+        :host             => host
+    )
+
+    bond.save!
+    host.interfaces.push bond
```

Comment:
Nit: This needs brackets...

Reply:
The call to push? Do you have a brackets-everywhere policy?

Reply:
(because then there would be more)